src/robotFactory.js

```diff
@@ -1,10 +1,76 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.coords = {
+      x: 0,
+      y: 0,
+      ...coords,
+    };
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
+  }
 
-class FlyingRobot {}
+  getInfo() {
+    const robot = `Robot: ${this.name}`;
+    const chipVersion = `Chip version: ${this.chipVersion}`;
+    const weight = `Weight: ${this.weight}`;
 
-class DeliveryDrone {}
+    return `${robot}, ${chipVersion}, ${weight}`;
+  };
+
+  goForward(steps = 1) {
+    this.coords.y += steps;
+  };
+
+  goBack(steps = 1) {
+    this.coords.y -= steps;
+  };
+
+  goRight(steps = 1) {
+    this.coords.x += steps;
+  };
+
+  goLeft(steps = 1) {
+    this.coords.x -= steps;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    if (coords.z === undefined) {
+      coords.z = 0;
+    }
```

Comment:
Simpler way to write this:
```suggestion
    this.coords.z = coords.z || 0;
```